File: src/entities/IncomeExpense.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, CreateDateColumn } from 'typeorm';
import { User } from './User';
import { Business } from './Business';

@Entity()
export class IncomeExpense {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  type: string; // income | expense

  @Column()
  amount: number;

  @Column()
  description: string;

  @ManyToOne(() => User, (user) => user.incomeExpenses)
  user: User;

  @ManyToOne(() => Business, (business) => business.incomeExpenses)
  business: Business;

  @CreateDateColumn()
  createdAt: Date;
} 

File: Entities/IncomeExpense.cs
using System;

namespace ModularSaaS.Entities
{
    public enum IncomeExpenseType
    {
        income,
        expense
    }

    public class IncomeExpense
    {
        public Guid Id { get; set; }
        public Guid BusinessId { get; set; }
        public Guid UserId { get; set; }
        public IncomeExpenseType Type { get; set; }
        public decimal Amount { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public Business Business { get; set; }
        public User User { get; set; }
    }
} 